### PR TITLE
chore: upgrade to node18

### PR DESCRIPTION
**Pull request by KEGustafsson**
This PR upgrade npm builds to use node18.
- npmjs package builds
- docker image builds
- docker base image build
- documentation references updated

In the same time docker base image is upgraded to use ubuntu22.04

### CODE REVIEW

**Comment by tkurki** [comment]
@sbender9 @panaaj Karl-Erik pinged about upgrading to Node 18 and I said why not - newer Nodes have usually performance advantages. He says he's been running on 18 himself for a while without problems. What do you think?

I also thought that we would do a "soft" update:
- start running tests in 18 but keep also 16, so that we don't accidentally commit to 18 without communicating that
- build docker with 18
- change installation instructions to refer to 18

and then only later(?) communicate that the official version is 18.

16 EOL schedule is 2023-09-11.

**Comment by sbender9** [comment]
Fine with me

**Comment by panaaj** [comment]
Also fine with me.
To make use of libraries like PMTiles to support charts served from cloud storage, node18 is required so you have my vote.

**Comment by tkurki** [comment]
Ping also @mpvader

**Comment by mpvader** [comment]
Thanks for the ping, appreciated. I expect to have an official version of Venus OS on nodejs 18 late(r) this year. Current state of that is here, few commits down. https://github.com/victronenergy/meta-victronenergy/commits/mbath/grafana

As long as its no big deal, indeed why not keep signalk-server also working on nodejs v16? Also other companies and initiatives that include signalk-server might need to some time to upgrade to v18.

**Comment by KEGustafsson** [comment]
I could modify this PR to add additional build actions for docker images based on 18 and keep existing build actions untouched until you feel ready to move 18.
E.g.
- `build-docker-18.yml` -> `signalk-server:node18-<tag>`
- `build-base-image-2204.yml` -> `signalk-server-base:node18-latest`

**Comment by tkurki** [comment]
Just browsed through the PR…docker we can switch over to 18, but let’s run tests on both 16 *and* 18 for the time being to assure compatibility.

Server release build can run with 18.

@panaaj we can prob make charts plugin continue to work with 16 even if pmtiles requires 18?

**Comment by panaaj** [comment]
The plugin will not break if node version is not 18, it's just that any PMTILES detected will throw an exception which is handled.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [16.x]
+        node-version: [18.x]
```
Comment (tkurki) [inline .github/workflows/test.yml]: Let's keep Node 16 here, as discussed in the PR.

**Review by tkurki — CHANGES_REQUESTED** [review]
- need to keep tests on 16 as well as 18
- this PR is based on an early version of ghcr and will cause a mess if merged - please remove the ghcr stuff, just the node version bump

File: .github/workflows/test.yml (hunk 1 above)
Comment (KEGustafsson) [reply .github/workflows/test.yml]: Test matrix updated `[16.x, 18.x]`

**Comment by KEGustafsson** [comment]
>     * this PR is based on an early version of ghcr and will cause a mess if merged - please remove the ghcr stuff, just the node version bump
Cant figure out why I made this PR based on the ghcr branch?!? Now it should be cleaned.

**Review by tkurki — APPROVED** [review]

**Pull request merged**